j1939/tst-j1939-ac.c

```diff
@@ -0,0 +1,1652 @@
+ * communicate not only errors, but also different state information about the
+ * status of egress or ingress packets, as well as timestamps related to these
+ * states (e.g., the actual transmission time to the CAN bus). If
+ * tst_j1939_ac_recv_err returns an error other than EINTR, the function returns
+ * the error code. Otherwise, it continues processing events.
+ *
+ * Returns: 0 on success, error code on failure
+ */
+static int tst_j1939_ac_handle_events(struct tst_j1939_ac_priv *priv,
+                                      int nfds) {
+        int ret;
+        int n;
+
+        for (n = 0; n < nfds && n < priv->epoll_events_size; ++n) {
+		struct epoll_event *ev = &priv->epoll_events[n];
+		struct tst_j1939_ac_sock *sock = ev->data.ptr;
+
+		if (!sock) {
+			warn("no sock");
+			continue;
+		}
+
+		if (!ev->events) {
+			warn("no events");
+			continue;
+		}
+
+		if (ev->events & EPOLLERR) {
+			struct tst_j1939_ac_err_msg emsg = {
+				.stats = &sock->stats,
+			};
+
+			ret = tst_j1939_ac_recv_err(sock->fd, &emsg);
+			if (ret && ret != -EINTR)
+				return ret;
+		}
+	}
+
+	return 0;
+}
+
+/*
+ * tst_j1939_ac_process_events_and_tasks: Process events reported by epoll_wait
+ *					  and run self-tests
+ *
+ * @param priv: Pointer to the tst_j1939_ac_priv structure holding test setup
+ * and state information
+ *
+ * This function prepares for events by calling tst_j1939_ac_prepare_for_events
+ * and provides the number of file descriptors with reported events (nfds). If
+ * any events are reported, it calls tst_j1939_ac_handle_events to process the
+ * events. The function then calls tst_j1939_ac_run_self_tests to run the
+ * self-tests.
+ *
+ * Returns: 0 on success, error code on failure
+ */
+static int tst_j1939_ac_process_events_and_tasks(struct tst_j1939_ac_priv *priv)
+{
+	int nfds = 0;
+	int ret;
+
+	ret = tst_j1939_ac_prepare_for_events(priv, &nfds);
+	if (ret)
+		return ret;
+
+	if (nfds > 0) {
+		ret = tst_j1939_ac_handle_events(priv, nfds);
+		if (ret)
+			return ret;
+	}
+
+	tst_j1939_ac_run_self_tests(priv);
+
+	return 0; 
+}
+
+/*
+ * tst_j1939_ac_epoll_prepare: Initialize epoll instance for the given private
+ *			       test structure
+ *
+ * @param priv: Pointer to the tst_j1939_ac_priv structure holding test setup
+ *		and state information
+ *
+ * This function initializes an epoll instance for the given private test
+ * structure by calling epoll_create1. On failure, it exits with an error
+ * message. It also sets the epoll_events_size based on the size of the
+ * epoll_events array.
+ */
+static void tst_j1939_ac_epoll_prepare(struct tst_j1939_ac_priv *priv) {
+        priv->epoll_fd = epoll_create1(0);
+        if (priv->epoll_fd < 0)
+                err(EXIT_FAILURE, "epoll_create1");
+
+        priv->epoll_events_size = ARRAY_SIZE(priv->epoll_events);
+}
+
+static void tst_j1939_ac_print_help(void)
+{
+	printf("Usage: tst-j1939-ac [options]\n");
```

Comment:
please format the help output so that it is usable with `help2man`, see https://github.com/linux-can/can-utils/pull/69